Networking/Networking/Remote/MediaRemote.swift

```diff
@@ -34,6 +37,36 @@ public protocol MediaRemoteProtocol {
 /// Media: Remote Endpoints
 ///
 public class MediaRemote: Remote, MediaRemoteProtocol {
+    /// Loads media from the site's WP Media library
+    /// API reference - https://developer.wordpress.org/rest-api/reference/media/#retrieve-a-media-item
+    /// 
+    /// - Parameters:
+    ///   - siteID: site ID for which to load Media
+    ///   - mediaID: ID of the Media to load
+    ///   - completion: Closure to be executed upon completion.
+    ///
+    public func loadMedia(siteID: Int64,
+                          mediaID: Int64,
+                          completion: @escaping (Result<WordPressMedia, Error>) -> Void) {
+        let parameters: [String: Any] = [
+            ParameterKey.fieldsWordPressSite: ParameterValue.wordPressMediaFields,
+        ].compactMapValues { $0 }
+
+        let path = "sites/\(siteID)/media/\(mediaID)"
+        do {
+            let request = try DotcomRequest(wordpressApiVersion: .wpMark2,
```

Comment:
Is there any reason for using the WPCom endpoint here instead of the endpoint of the WP Rest API `/wp/v2/media`? Using the Core endpoint will make this possible for all logins, even if we don't need it right now. And since both endpoints return the same response, the change will be easy hopefully.

Reply:
Great observation, @hichamboushaba! 

This will work for application password logins as well. 

I understand that the name `DotcomRequest` might have caused confusion, but the parameter `availableAsRESTRequest`, which is passed as `true`, will take care of making this request work for application password logins. You can take a look at the code [here](https://github.com/woocommerce/woocommerce-ios/blob/trunk/Networking/Networking/Requests/DotcomRequest.swift#L112-L132).

Kindly let me know if you have further questions.

Reply:
Oh, interesting, thanks for the clarification @selanthiraiyan